components/calendar/src/hebrew.rs

```diff
@@ -0,0 +1,1151 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! This module contains types and implementations for the Civil Hebrew calendar.
+//!
+//! ```rust
+//! use icu::calendar::{Date, DateTime};
+//!
+//! // `Date` type
+//! let hebrew_date = Date::try_new_civil_hebrew_date(3425, 10, 11)
+//!     .expect("Failed to initialize hebrew Date instance.");
+//!
+//! // `DateTime` type
+//! let hebrew_datetime = DateTime::try_new_civil_hebrew_datetime(3425, 10, 11, 13, 1, 0)
+//!     .expect("Failed to initialize hebrew DateTime instance.");
+//!
+//! // `Date` checks
+//! assert_eq!(hebrew_date.year().number, 3425);
+//! assert_eq!(hebrew_date.month().ordinal, 10);
+//! assert_eq!(hebrew_date.day_of_month().0, 11);
+//!
+//! // `DateTime` checks
+//! assert_eq!(hebrew_datetime.date.year().number, 3425);
+//! assert_eq!(hebrew_datetime.date.month().ordinal, 10);
+//! assert_eq!(hebrew_datetime.date.day_of_month().0, 11);
+//! assert_eq!(hebrew_datetime.time.hour.number(), 13);
+//! assert_eq!(hebrew_datetime.time.minute.number(), 1);
+//! assert_eq!(hebrew_datetime.time.second.number(), 0);
+//! ```
+
+use crate::calendar_arithmetic::{ArithmeticDate, CalendarArithmetic};
+use crate::helpers::{self, div_rem_euclid, div_rem_euclid_f64, final_func, next_u8};
+use crate::julian::Julian;
+use crate::rata_die::RataDie;
+use crate::types::Moment;
+use crate::Iso;
+use crate::{types, Calendar, CalendarError, Date, DateDuration, DateDurationUnit, DateTime};
+use ::tinystr::tinystr;
+
+/// Biblical Hebrew Calendar
+#[derive(Copy, Clone, Debug, Default, Hash, Eq, PartialEq, PartialOrd, Ord)]
+#[allow(clippy::exhaustive_structs)]
+pub struct BookHebrew {
+    year: i32,
+    month: u8,
+    day: u8,
+}
+/// Civil Hebrew Calendar
+#[derive(Copy, Clone, Debug, Default, Hash, Eq, PartialEq, PartialOrd, Ord)]
+#[allow(clippy::exhaustive_structs)]
+pub struct CivilHebrew;
```

Comment:
nit: I think the calendar should still be called `Hebrew` (also the DateInner), at least from the perspective of public code